File: vue-python/4-backend/app/schemas/ai.py
from pydantic import BaseModel, Field


class ProcurementFields(BaseModel):
    """LLM 返回的候选字段；所有值都必须由用户进一步确认。"""

    material: str | None = None
    specification: str | None = None
    quantity: int | None = Field(default=None, ge=1)
    unit: str | None = None
    expectedDeliveryDate: str | None = None
    purpose: str | None = None
    preferredBrand: str | None = None
    preferredSupplier: str | None = None


class AIExtractionRequest(BaseModel):
    message: str = Field(min_length=1, max_length=4_000)
    clientRequestId: str = Field(min_length=1, max_length=128)
    expectedVersion: int = Field(ge=0)


class AIExtractionResponse(BaseModel):
    jobId: str
    status: str = "SUCCEEDED"
    fields: ProcurementFields
    missingFields: list[str] = Field(default_factory=list)
    ambiguities: list[str] = Field(default_factory=list)

File: vue-python/3-frontend/src/api/ai.ts
import { api } from './client'

export interface ProcurementFields {
  material: string | null
  specification: string | null
  quantity: number | null
  unit: string | null
  expectedDeliveryDate: string | null
  purpose: string | null
  preferredBrand: string | null
  preferredSupplier: string | null
}

interface ExtractionResult {
  jobId: string
  status: 'SUCCEEDED'
  fields: ProcurementFields
  missingFields: string[]
  ambiguities: string[]
}

export async function extractProcurementFields(draftId: string, message: string): Promise<ExtractionResult> {
  const { data } = await api.post<{ data: ExtractionResult }>(`/drafts/${draftId}/ai-extractions`, {
    message,
    clientRequestId: crypto.randomUUID(),
    expectedVersion: 0,
  })
  return data.data
}
